src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -1413,7 +1352,7 @@ COOP_PINVOKE_HELPER(void, RhpReversePInvoke, (ReversePInvokeFrame * pFrame))
     if (pCurThread->InlineTryFastReversePInvoke(pFrame))
         return;
 
-    RhpReversePInvokeAttachOrTrapThread2(pFrame);
+    pCurThread->ReversePInvokeAttachOrTrapThread(pFrame);
```

Comment:
Is the C++ code generated for this going to be as good as before? It looks like this was carefully setup to allow the slow path to be tailcalled.

Reply:
The slow path would be very rare. It handles a case when a thread has not seen managed code yet. 
I thought that the helper was added to call from assembly and now we don't

I will check if we are still tailcalling.

Reply:
c++ actually inlines `RhpReversePInvokeAttachOrTrapThread2` after the change. 
The impact on the fast part of the method is not too bad, but keeping the slow path separate still has some minor benefits like pushing fewer registers in prolog. 

I will revert this part.